### DISCUSSION TITLE
How many devices are supported with CC2652R(B)?

### DISCUSSION THREAD

**Discussion by okastl** [Koenkk/zigbee2mqtt#8037] [General]
Hi there,
I am currently running a Zigbee network with almost 60 devices. Mostly bulbs (routers), but probably 20 end devices like switches, smoke detectors, sensors.
If I continue with my current pace, I will hit 100 devices soon. 15 Sonoff ZBMINI and 15 QS-Zigbee-C01 curtain modules are planned. Maybe another bulb or wall switch here and there...
Here https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator is said, that the stick can have 50 direct children and that 100/200 routes are possible. I do not understand, how the "100/200" routes limitation relates to the number of devices allowed in the network.
The notes say: "100 normal routes and 200 source routes can be kept in memory". Do I have any influence on which type of route is used? Does an additional device require 1 route? Or maybe 2? Or are the allowed routes and the numbers of devices not directly related?
I have searched, but I couldn't find a definite answer. Any insight is welcome. Thanks in advance.

**Comment by Koenkk** [comment]
In order for the coordinator to send a message to a device a route is required (exactly one). If it doesn't have one yet it obtains one from the network (this takes some time, a few milliseconds). The routing table (100/200 routes) acts as a cache for routes, so it does not have to query a route again when you try to send a command for a device, this is much more efficient of course. If the routing table is full and you send a command to a device which has no known route yet, the coordinator will discard the oldest route.

A route is only needed for device you control (e.g. switches, bulbs) and not for e.g. temperature sensors.

**Reply by okastl** [reply]
Thank you for the explanation. Makes me want to spend some money on door/window sensors.
One question remains: When is source routing (max. 200 routes) used? Who or what decides it?

**Reply by Koenkk** [reply]
Always (since those have all hops in it) normal routes only contains the first hop
